client/v3/retry_interceptor.go

```diff
@@ -297,17 +298,26 @@ func waitRetryBackoff(ctx context.Context, attempt uint, callOpts *options) erro
 }
 
 // isSafeRetry returns "true", if request is safe for retry with the given error.
-func isSafeRetry(lg *zap.Logger, err error, callOpts *options) bool {
+func isSafeRetry(c *Client, err error, callOpts *options) bool {
 	if isContextError(err) {
 		return false
 	}
+
+	// Situation when learner refuses RPC its supposed to not serve is from server
```

Comment:
Yea. We should have a more reliable way for client to retry on the learner case. The client should be able to tell if the server instance is a leaner or not.

Reply:
Added TODO